packages/core/src/FormField.js

```diff
@@ -1,235 +1,78 @@
 import React from 'react'
 import PropTypes from 'prop-types'
-import styled, { keyframes } from 'styled-components'
-import Box from './Box'
 import Flex from './Flex'
-import Select from './Select'
-import Icon from './Icon'
-import Label from './Label'
-import Input from './Input'
-import theme from './theme'
-
-const Root = styled(Box)`
-  & ${Box} {
-    pointer-events: none;
-  }
-`
-
-const fadeIn = keyframes`
-  from {
-    opacity: 0;
-  }
-
-  to {
-    opacity: 1;
-  }
-`
-const labelStyles = {
-  animation: fadeIn + ' 0.3s'
-}
-
-const getFieldStyles = showLabel => {
-  return showLabel
-    ? {
-        paddingTop: '20px',
-        paddingBottom: '8px',
-        transition: 'padding-top 0.1s, padding-bottom 0.1s'
-      }
-    : {
-        paddingTop: '14px',
-        paddingBottom: '14px',
-        transition: 'padding-top 0.1s, padding-bottom 0.1s'
+import IconField from './IconField'
+
+const FormField = props => {
+  let iconBefore = false
+
+  const children = React.Children.toArray(props.children)
+  const [field] = children.filter(child => child.type.isField)
+  const [label] = children.filter(child => child.type.isLabel)
+  const showLabel =
+    (label && !label.props.autoHide) || (field && !!field.props.value)
+
+  const styled = children.map((child, i, arr) => {
+    if (child.type.isField && arr[i - 1] && arr[i - 1].type.isIcon)
+      iconBefore = true
+    if (child === field) {
+      return React.cloneElement(child, {
+        style: {
+          ...child.props.style,
+          transitionProperty: 'padding-top, padding-bottom',
+          transitionDuration: '.1s',
+          paddingTop: showLabel ? 20 : 14,
+          paddingBottom: showLabel ? 8 : 14
+        }
+      })
+    }
+    return child
+  })
+
+  const styledLabel =
+    label &&
+    React.cloneElement(label, {
```

Comment:
This should also automatically apply the `htmlFor` prop like the current FormField